### DISCUSSION TITLE
Subfolder index (TOC)

### DISCUSSION THREAD

**Discussion by alkisg** [mkdocs/mkdocs#2671] [Q&A] [answered]
Hi, I'm a new mkdocs user. I want to include an index, e.g. a list of links to `advanced-docs/*.md`, in two places:

1) In my sidebar navigation, and
2) Inside the `advanced-docs/index.md` page (e.g. by specifying [toc: *.md] or something similar).

See an example: https://ltsp.org/advanced/ - currently I have to manually update that list.

For (1), I think some plugin exists, but I haven't found anything for (2), some help please?

I've read that "mkdocs has no notion of folders or hierarchy", but surely it has some way to generate a TOCs via globs, right?

**Comment by oprypin** [comment, marked as answer]
https://github.com/oprypin/mkdocs-literate-nav lets you write the nav in Markdown directly and then copy it over to the actual nav. I really recommend that because then your navs are readable also on GitHub or whatever other Git Web interface it might be. *And* you will probably want to control the order of the nav anyway, so writing them out may be unavoidable.

That plugin also has glob features, but the globs won't be expanded in Markdown, only in the actual nav, so unfortunately it's an either-or situation. Could be a nice enhancement to the plugin, though; I haven't thought about it.

A more typical recommendation just for globs is, of course, https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin/

And the most direct solution for actually populating Markdown *from* the nav seems to be https://github.com/Andre601/mkdocs-pagenav-generator - haven't tried it, though.

**Reply by Andre601** [reply]
I personally use the awesome pages plugin with my own and the result can be seen here: https://www.andre601.ch/blog/posts

The folder structure itself here is
```
docs/
 |
 |- posts/
     |
     |- .pages.yml
     |- index.md
     |- 2020/
     |- 2021/
```

**Comment by alkisg** [comment]
Thank you both! I will look into these shortly.
